### DISCUSSION TITLE
Rtools

### DISCUSSION THREAD

**Discussion by twa1970** [metrumresearchgroup/mrgsolve#888] [Q&A] [answered]
Dear All

I am new to mrgsolve and have some problems installing Rtools on my corporate computer. I have installed Rtools version 4.0, R version 4.1.2 and Rstudio 2021.09.1 Build 372. 

Following installation of  Rtools I ran the code "write('PATH="${RTOOLS40_HOME}\\usr\\bin;${PATH}"', file = "~/.Renviron", append = TRUE)" and the file is updated correctly.  Sys.which("make") returns following:
                              make 
"C:\\rtools40\\usr\\bin\\make.exe" 

But when I run the: install.packages("jsonlite", type = "source") as additional test I get following:

Installing package into ‘C:/Users/TWA/Documents/R/win-library/4.1’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/jsonlite_1.7.2.tar.gz'
Content type 'application/x-gzip' length 421716 bytes (411 KB)
downloaded 411 KB

* installing *source* package 'jsonlite' ...
** package 'jsonlite' successfully unpacked and MD5 sums checked
** using staged installation
** libs
"C:/rtools40/mingw64/bin/"gcc  -I"C:/PROGRA~1/R/R-41~1.2/include" -DNDEBUG -Iyajl/api       -D__USE_MINGW_ANSI_STDIO   -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c base64.c -o base64.o
      1 [main] make (4536) C:\rtools40\usr\bin\make.exe: *** fatal error - cygheap base mismatch detected - 0x13B2408/0x15F2408.
This problem is probably due to using incompatible versions of the cygwin DLL.
Search for cygwin1.dll using the Windows Start->Find/Search facility
and delete all but the most recent version.  The most recent version *should*
reside in x:\cygwin\bin, where 'x' is the drive on which you have
installed the cygwin distribution.  Rebooting is also suggested if you
are unable to find another cygwin DLL.
      0 [main] make 1288 dofork: child -1 - forked process 4536 died unexpectedly, retry 0, exit code 0xC0000142, errno 11
make: fork: Resource temporarily unavailable
ERROR: compilation failed for package 'jsonlite'
* removing 'C:/Users/TWA/Documents/R/win-library/4.1/jsonlite'
* restoring previous 'C:/Users/TWA/Documents/R/win-library/4.1/jsonlite'
Warning in install.packages :
  installation of package ‘jsonlite’ had non-zero exit status

I have also tried to set up R, Rtools and Rstudio on my private compute where is all works - Do you have any suggestions for how to fix this on my corporate computer?

Thanks

Br 

Thomas Anderson

**Comment by kylebaron** [comment]
Hi @twa1970 - 

I haven't seen this before. Wondering if you did find another `cygwin` distribution installed?

```
*** fatal error - cygheap base mismatch detected - 0x13B2408/0x15F2408.
This problem is probably due to using incompatible versions of the cygwin DLL.
Search for cygwin1.dll using the Windows Start->Find/Search facility
and delete all but the most recent version. The most recent version should
reside in x:\cygwin\bin, where 'x' is the drive on which you have
installed the cygwin distribution. Rebooting is also suggested if you
are unable to find another cygwin DLL.
```

Kyle

**Comment by twa1970** [comment]
Hi Kyle 

Thanks for looking into this. No, I can't see any other cygwin. When I press Start -> Serach "cygwin" I only get this file: 
" C:\Users\TWA\Documents\R\win-library\4.1\BH\include\boost\config\platform\cygwin.hpp"

Br

Thomas

**Comment by kylebaron** [comment]
Thanks for checking on that. It does seem like rtools is installed properly but something in the operating system doesn't want to deal with the `.dll` that is generated. I googled this a bit and it seems like there have been reports of this before (for example):

https://github.com/r-lib/rlang/issues/232#issuecomment-337875941

But this does seem rather old and it's hard to tell if there is an obvious solution somewhere in there. 

One last ditch thing that I might try is (physically) remove all of your installed packages and try again just to make sure you're really starting from scratch. Seems like a long shot and wouldn't be surprised if you get the same thing. But maybe worth a try. 

One other test I'm curious about: if you install all of your packages as Windows binaries, can you still compile a model for mrgsolve?

```r
install.packages("mrgsolve")
library(mrgsolve)
mod <- modlib("pk1")
```

**Comment by twa1970** [comment, marked as answer]
Hi Kyle

I have solved the this issue. 

Problem solved by setting Forced randomization for images (Mandatory ASLR) in Windows Security to "On by default"

@twa1970
